### ISSUE TITLE
Convert OnSwipeTouchListener to kotlin

### ISSUE THREAD

**Issue by gouri-panda** [kiwix/kiwix-android#2400]
Is your feature request related to a problem? Please describe.

OnSwipeTouchListener is in Java

Describe the solution you'd like

Convert it in Kotlin

Additional context

Parent issue #1826

**Comment by abhijeetsinghcode** [comment]
This is right 
import android.annotation.SuppressLint
import android.content.Context
import android.view.GestureDetector
import android.view.MotionEvent
import android.view.View
import kotlin.math.abs

open class OnSwipeTouchListener : View.OnTouchListener {
    private var gestureDetector: GestureDetector

    @SuppressLint("SyntheticAccessor")
    constructor(c: Context?) {
        gestureDetector = GestureDetector(c, GestureListener())
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouch(v: View, event: MotionEvent): Boolean {
        return gestureDetector.onTouchEvent(event)
    }
    private inner class GestureListener : GestureDetector.SimpleOnGestureListener() {
        private val SWIPE_THRESHOLD: Int = 100
        private val SWIPE_VELOCITY_THRESHOLD: Int = 100
        override fun onDown(e: MotionEvent): Boolean {
            return true
        }
        override fun onSingleTapUp(e: MotionEvent): Boolean {
            onTap(e)
            return super.onSingleTapUp(e)
        }

        override fun onFling(e1: MotionEvent, e2: MotionEvent, velocityX: Float, velocityY: Float): Boolean {
            var result: Boolean = false
            try {
                val diffY = e2.y - e1.y
                val diffX = e2.x - e1.x
                if (abs(diffX) > abs(diffY)) {
                    if (abs(diffX) > SWIPE_THRESHOLD && abs(
                            velocityX
                        ) > SWIPE_VELOCITY_THRESHOLD
                    ) {
                        if (diffX > 0) {
                            onSwipeRight()
                        }
                        else {
                            onSwipeLeft()
                        }
                        result = true
                    }
                }
                else if (abs(diffY) > SWIPE_THRESHOLD && abs(
                            velocityY
                        ) > SWIPE_VELOCITY_THRESHOLD
                    ) {
                        if (diffY > 0) {
                            onSwipeBottom()
                        }
                        else {
                            onSwipeTop()
                        }
                        result = true
                    }

            } catch (exception: Exception) {
                exception.printStackTrace()
            }
            return result
        }
    }
    open fun onSwipeRight() {

    }
    open fun onSwipeLeft() {

    }
    open fun onSwipeTop() {

    }
    open fun onSwipeBottom() {

    }
    open fun onTap(e: MotionEvent) {

    }

}

**Comment by abhijeetsinghcode** [comment]
Above code I checked on app , I don't know how to test the code in main application.

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.widget.RelativeLayout
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
class MainActivity : AppCompatActivity() {
    private lateinit var layout: RelativeLayout
    @SuppressLint("ClickableViewAccessibility")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        layout = findViewById(R.id.relativeLayout)
        layout.setOnTouchListener(object : OnSwipeTouchListener(this@MainActivity) {

            override fun onTouch(v: View, event: MotionEvent): Boolean {
                
                return super.onTouch(v, event)
            }

            override fun onSwipeTop() {
                super.onSwipeTop()
                Toast.makeText(this@MainActivity, "Swipe Top gesture detected",
                    Toast.LENGTH_SHORT)
                    .show()
            }

            override fun onSwipeBottom() {
                super.onSwipeBottom()
                Toast.makeText(this@MainActivity, "Swipe Bottom gesture detected",
                    Toast.LENGTH_SHORT)
                    .show()
            }

            override fun onTap(e: MotionEvent) {
                super.onTap(e)
                Toast.makeText(this@MainActivity, "Tap gesture detected",
                    Toast.LENGTH_SHORT)
                    .show()
            }

            override fun onSwipeLeft() {
                super.onSwipeLeft()
                Toast.makeText(this@MainActivity, "Swipe Left gesture detected",
                    Toast.LENGTH_SHORT)
                    .show()
            }
            override fun onSwipeRight() {
                super.onSwipeRight()
                Toast.makeText(
                    this@MainActivity,
                    "Swipe Right gesture detected",
                    Toast.LENGTH_SHORT
                ).show()
            }


        })
    }
}

**Comment by gouri-panda** [comment]
TL;DR. @abhijeetsinghcode this is not an ideal way to solve an issue. If you want to solve this issue solve it and then make a pull request to the project. There we will review it. Do you wanna work on this issue?

**Comment by abhijeetsinghcode** [comment]
Sorry, I will try to make  a pull request for this.

**Comment by gouri-panda** [comment]
>Do you wanna work on this issue?

?

**Comment by gouri-panda** [comment]
@abhijeetsinghcode you are assigned!
